lib/Twig/Extension/Core.php

```diff
@@ -821,11 +821,27 @@ function _twig_default_filter($value, $default = '')
 function twig_get_array_keys_filter($array)
 {
     if ($array instanceof Traversable) {
-        return array_keys(iterator_to_array($array));
-    }
+        if ($array instanceof IteratorAggregate) {
+            $array = $array->getIterator();
+        }
 
-    if (!is_array($array)) {
```

Comment:
ah, must how thrown it away when I removed the deprecation warning, fixed, thanks